csvqb/csvqb/tests/unit/pydantic/test_cube.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import pytest
+
+from csvqb.models.cube import *
+from csvqb.tests.unit.test_baseunit import assert_num_validation_errors
+
+
+def test_attribute_property_validation():
+    """Testing that the pydantic validation does deep validation of a model."""
+    metadata = CatalogMetadata("Some Qube")
+    data = pd.DataFrame({"A": ["a", "b", "c"], "Value": [1, 2, 3]})
+    columns = [
+        QbColumn(
+            "A",
+            NewQbDimension(
+                "Some New Dimension",
+                code_list=NewQbCodeList(
+                    CatalogMetadata("Some Code List"),
+                    # N.B. The Concepts shouldn't be strings, this should cause a validation error
+                    concepts=["Hello", "World"],
+                ),
+            ),
+        ),
+        QbColumn(
+            "Value",
+            QbSingleMeasureObservationValue(
+                NewQbMeasure("Some Measure"), NewQbUnit("Some Unit")
+            ),
+        ),
+    ]
+
+    cube = Cube(metadata, data, columns)
+    errors = cube.validate()
+    assert_num_validation_errors(errors, 2)
```

Comment:
I now see why you've created this. Query withdrawn.

Reply:
Yeah, it's because I got fed up debugging to figure out what the unexpected errors were telling me.